src/libraries/System.Security.Cryptography.Cose/src/System/Security/Cryptography/Cose/CoseMessage.cs

```diff
@@ -60,6 +74,14 @@ internal CoseMessage(CoseHeaderMap protectedHeader, CoseHeaderMap unprotectedHea
         [MemberNotNullWhen(false, nameof(Content))]
         internal bool IsDetached => _content == null;
 
+        /// <summary>
+        /// Decodes a CBOR payload as a COSE_Sign1 message.
+        /// </summary>
+        /// <param name="cborPayload">The sequence of CBOR-encoded bytes to decode.</param>
+        /// <returns>A <see cref="CoseSign1Message"/> representation of <paramref name="cborPayload"/>.</returns>
+        /// <exception cref="ArgumentNullException"><paramref name="cborPayload"/> is <see langword="null"/>.</exception>
+        /// <exception cref="ArgumentException"><paramref name="cborPayload"/> contains trailing data after COSE_Sign1 was complete.</exception>
```

Comment:
I'm having trouble deciding if I like it more, or less, that "the bytes didn't look like a COSE_Sign1" and "the bytes didn't look like a COSE_Sign1 because they had trailing data" are different.

Generally, an ArgumentException represents something the caller should have avoided doing, or should be able to check themselves.  In this case, I don't think they can reasonably be expected to check that, so I think I'd just call both cases a CryptographicException.

Reply:
Yes, in fact, this "trailing data" exception can be removed as it is encompassed on `<exception cref="CryptographicException"><paramref name="cborPayload"/> could not be decoded as a COSE_Sign1 message.</exception>`